include/engine/datafacade/contiguous_internalmem_datafacade.hpp

```diff
@@ -1070,12 +1070,46 @@ template <> class ContiguousInternalMemoryAlgorithmDataFacade<MLD> : public Algo
         InitializeInternalPointers(allocator->GetLayout(), allocator->GetMemory());
     }
 
-    const partition::MultiLevelPartitionView &GetMultiLevelPartition() const override
+    LevelID GetQueryLevel(NodeID source, NodeID target, NodeID node) const override
     {
-        return mld_partition;
+        return mld_partition.GetQueryLevel(source, target, node);
     }
 
-    const partition::CellStorageView &GetCellStorage() const override { return mld_cell_storage; }
+    LevelID GetHighestDifferentLevel(NodeID first, NodeID second) const override
+    {
+        return mld_partition.GetHighestDifferentLevel(first, second);
+    }
+
+    void ForEachSourceNodes(LevelID level, NodeID node, NodeFnT const &fn) const override
```

Comment:
@deniskoronchik sorry, i don't understand the use case, but what do you mean by the plain structure of data? The callback function signature is fixed to `void(NodeID, EdgeWeight, EdgeDuration)>` at compile time, but is it really needed to change a callback function at runtime and what would a use case for this?

As a solution i would suggest to add simple proxy-functions to facade (this requires explicit types for return values of function  and extend facade with `ForEachSourceNodes`/`ForEachDestinationNodes` to have possibility for runtime selection of a callback.